sdk/file_sdk/XMPFileFormat/xmp_helper.cpp

```diff
@@ -79,12 +123,51 @@ const vector<Tag> XMPHelper::GetTags(shared_ptr<IStream> fileStream) {
   catch(XMP_Error error)
   {
     file.CloseFile();
-    throw std::runtime_error(error.GetErrMsg());
+    throw Exception(error.GetErrMsg());
+  }
+  catch(...)
+  {
+    file.CloseFile();
+    throw Exception("GetXMP failed");
+  }
+}
+
+void XMPHelper::SetTags(shared_ptr<IStream> inputStream, shared_ptr<IStream> outputStream, const vector<Tag>& tags)
+{
+  XMPIOOverIStream xmpStream (inputStream);
+
+  SXMPFiles file;
+  if (!file.OpenFile(&xmpStream , kXMP_UnknownFile, kXMPFiles_OpenForUpdate | kXMPFiles_OpenUseSmartHandler) &&
+      !file.OpenFile(&xmpStream , kXMP_UnknownFile, kXMPFiles_OpenForUpdate | kXMPFiles_OpenUsePacketScanning))
+    throw Exception("OpenFile failed");
+
+  try
+  {
+    SXMPMeta metadata;
+    file.GetXMP(&metadata);
+    Serialize(metadata, tags);
+
+    if (!file.CanPutXMP(metadata))
+    {
+      file.CloseFile();
+      throw Exception("CanPutXMP failed");
+    }
+
+    inputStream->Seek(0);
+    file.PutXMP(metadata);

```

Comment:
doesn't this affect the inputStream?

Reply:
does it matter? the input stream is affected anyway

Reply:
Won't this change the input stream?

Reply:
The data in it

Reply:
The original input stream is opened for read only, that is why I need to copy the stream into fstream and this is the stream